content/courses/advanced-ada/parts/resource_management/memory_management.rst

```diff
@@ -6,9 +6,209 @@ Memory Management
 Maximum allocation size and alignment
 -------------------------------------
 
-.. todo::
+We've seen details about the :ada:`Size` and :ada:`Object_Size` attributes in the section about :ref:`data representation <Adv_Ada_Data_Representation_Sizes>`. :ref:`Later on <Adv_Ada_Storage_Size_Attribute>`, we also mentioned the :ada:`Storage_Size` attribute.
 
-    Complete section!
+In this section, we expand our discussion on sizes and talk about the :ada:`Max_Size_In_Storage_Elements` and the :ada:`Max_Alignment_For_Allocation` attributes. These attributes return values that are important in the allocation of :ref:`memory subpools <Adv_Ada_Memory_Pools>` via the :ada:`Allocate` procedure from the :ada:`System.Storage_Pools.Subpools` package:
+
+.. code-block:: ada
+
+    procedure Allocate (
+       Pool                     : in out Root_Storage_Pool_With_Subpools;
+       Storage_Address          :    out Address;
+       Size_In_Storage_Elements :        Storage_Elements.Storage_Count;
+       Alignment                :        Storage_Elements.Storage_Count);
+
+In fact, the :ada:`Max_Size_In_Storage_Elements` attribute indicates the maximum value that can be used for the actual :ada:`Size_In_Storage_Elements` parameter of the :ada:`Allocate` procedure . Likewise, the :ada:`Max_Alignment_For_Allocation` attribute indicates the maximum value for the actual :ada:`Alignment` parameter of the :ada:`Allocate` procedure. (We discuss more details about this procedure later on.)
+
+The :ada:`Allocate` procedure is called when we allocate memory for access types. Therefore, the value returned by the :ada:`Max_Size_In_Storage_Elements` attribute for a subtype :ada:`S` indicates the maximum value of storage elements when allocating memory for an access type whose designated subtype is :ada:`S`, while the :ada:`Max_Alignment_For_Allocation` attribute indicates the maximum alignment that we can use for the allocation.
```

Comment:
May want to say "when we allocate memory via the :ada:`new` allocator."

Reply:
Yes, that sounds better.